### PR TITLE
Fix visit of true and false on Rails 4.2+

### PR DESCRIPTION
**Pull request by pschambacher**
🚂 

/cc @zendesk/classic-upgrade

### CODE REVIEW

File: lib/kasket/visitor.rb [hunk 1]
```diff
@@ -94,8 +94,13 @@ def visit_Arel_Nodes_In(node, *_)
     end
 
     def visit_Arel_Nodes_Equality(node, *_)
-      right = node.right
-      [visit(node.left), right ? visit(right) : nil]
+      right = case node.right
+      when false then 0 # This should probably be removed when Rails 3.2 is not supposed anymore
```
Comment (bquorning) [inline lib/kasket/visitor.rb]: `s/supposed/supported/`

File: lib/kasket/configuration_mixin.rb [hunk 2]
```diff
@@ -52,7 +52,11 @@ def kasket_key_for_id(id)
     end
 
     def kasket_indices
-      result = @kasket_indices || []
+      result = if defined?(@kasket_indices) && @kasket_indices
```
Comment (grosser) [inline lib/kasket/configuration_mixin.rb]: `@kasket_indices ||= []` won't work ?

Comment (pschambacher) [inline lib/kasket/configuration_mixin.rb]: It *works* but generates so much ruby warnings that the output is unreadable 😄

Comment (grosser) [inline lib/kasket/configuration_mixin.rb]: `||=` does not produce warnings

**Pull request merged**